Add plugin: TreeFocus

iOSonntag opened the pull request:
# I am submitting a new Community Plugin
## NOTE
After review is performed and you think I could be a great fit for the **Obsidian** team - please let me know! 

I would be greatly interested and would be happy to send a real application.

## Obsidian Modifications
- [x] I use the `debounce()` method when applying my changes
- [x] I store plugin settings in 
  - [x] the plugins own `data.json`
  - [x] and they can be reset in the settings tab
- [x] I store additional data in 
  - [x]  the plugins own `data.json`
  - [ ]  another file:  `RELATIVE_PATH_FROM_PLUGIN`
  - [ ]  another location:  `RELATIVE_PATH_FROM_PLUGIN`
- [x] I changed obsidian style & appearance with my `styles.css`
  - [x] I added custom classes
    - [x] the custom classes tend to follow naming conventions from **Obsidian** classes
  - [x] I am mostly referencing variables e.g. `var(--text-accent)`
  - [x] I created custom native values e.g. `3px`, 
  but I am sure they can **NOT** be derived from existing **Obsidian** variables
  
*PS: If it brings you any value - you can copy the list.*

### Additional modifications worth mentioning
- adding custom html data attributes to existing file explorer items 
e.g. `data-treefocus-mode="HIGHLIGHT"`
- compatible with `obsidian-iconize` plugin (former `obsidian-icon-folder`)

## Repo URL

Link to my plugin: [TreeFocus - GitHub Repository](https://github.com/iOSonntag/obsidian-plugin-treefocus)

## Release Checklist
- [x] I have tested the plugin on
  - [ ]  Windows
  - [x]  macOS
  - [ ]  Linux
  - [ ]  Android _(if applicable)_
  - [ ]  iOS _(if applicable)_
- [x] My GitHub release contains all required files
  - [x] `main.js`
  - [x] `manifest.json`
  - [x] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the tips in https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

joethei commented:
["repository": {](https://github.com/iOSonntag/obsidian-plugin-treefocus/blob/c28d9a29549950b2d8729b668f023e6050d672fc/manifest.json#L10-L16)
This section is not part of the manifest spec, remove it.
<https://docs.obsidian.md/Reference/Manifest>

['Transformation Rules'](https://github.com/iOSonntag/obsidian-plugin-treefocus/blob/c28d9a29549950b2d8729b668f023e6050d672fc/src/views/settings-view.ts#L72)
[.setButtonText('Reset to Defaults')](https://github.com/iOSonntag/obsidian-plugin-treefocus/blob/c28d9a29549950b2d8729b668f023e6050d672fc/src/views/settings-view.ts#L79)
[.setButtonText('Add Rule')](https://github.com/iOSonntag/obsidian-plugin-treefocus/blob/c28d9a29549950b2d8729b668f023e6050d672fc/src/views/settings-view.ts#L86)
and more
[Use sentence case in UI](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Use+sentence+case+in+UI)

[if (process.env.NODE_ENV === 'production') return;](https://github.com/iOSonntag/obsidian-plugin-treefocus/blob/c28d9a29549950b2d8729b668f023e6050d672fc/src/util/logger.ts#L17)
This will crash the plugin on mobile, since there is no Node there.

iOSonntag commented:
@joethei thanks for your review. I fixed all of the mentioned issues.

Regarding the third one - I added a `Platform.isDesktopApp` check and applied questions marks on the every var like `process?.env?.NODE_ENV` this should serve the case right?
